src/Storage/Storage.Management/File/UpdateAzureStorageShare.cs

```diff
@@ -0,0 +1,168 @@
+        private const string AccountNameParameterSet = "AccountName";
+
+        /// <summary>
+        /// Account object parameter set 
+        /// </summary>
+        private const string AccountObjectParameterSet = "AccountObject";
+
+        /// <summary>
+        /// ShareObject Parameter Set
+        /// </summary>
+        private const string ShareObjectParameterSet = "ShareObject";
+
+        /// <summary>
+        /// Share ResourceId  parameter set 
+        /// </summary>
+        private const string ShareResourceIdParameterSet = "ShareResourceId";
+
+        [Parameter(
+            Position = 0,
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Resource Group Name.",
+            ParameterSetName = AccountNameParameterSet)]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(
+            Position = 1,
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Storage Account Name.",
+            ParameterSetName = AccountNameParameterSet)]
+        [Alias(AccountNameAlias)]
+        [ValidateNotNullOrEmpty]
+        public string StorageAccountName { get; set; }
+
+        [Alias("N", "ShareName")]
+        [Parameter(Mandatory = true,
+            HelpMessage = "Share Name",
+            ValueFromPipeline = true,
+            ValueFromPipelineByPropertyName = true,
+            ParameterSetName = AccountObjectParameterSet)]
+        [Parameter(Position = 2,
+            Mandatory = true,
+            HelpMessage = "Share Name",
+            ValueFromPipeline = true,
+            ValueFromPipelineByPropertyName = true,
+            ParameterSetName = AccountNameParameterSet)]
+        public string Name { get; set; }
+
+        [Parameter(Mandatory = true,
+            HelpMessage = "Storage account object",
+            ValueFromPipeline = true,
+            ValueFromPipelineByPropertyName = true,
+            ParameterSetName = AccountObjectParameterSet)]
+        [ValidateNotNullOrEmpty]
+        public PSStorageAccount StorageAccount { get; set; }
+
+        [Parameter(
+            Position = 0,
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Input a File Share Resource Id.",
+           ParameterSetName = ShareResourceIdParameterSet)]
+        [ValidateNotNullOrEmpty]
+        public string ResourceId { get; set; }
+
+        [Alias("Share")]
+        [Parameter(Mandatory = true,
+            HelpMessage = "Storage Share object",
+            ValueFromPipeline = true,
+            ValueFromPipelineByPropertyName = true,
+            ParameterSetName = ShareObjectParameterSet)]
+        [ValidateNotNullOrEmpty]
+        public PSShare InputObject { get; set; }
+
+        [Parameter(Mandatory = false,
+            HelpMessage = "Share Quota in Gibibyte.")]
+        public int QuotaGiB
```

Comment:
QuotaGB or QuotaInGB

Reply:
See above comments, GiB is different from GB, and this should be GiB. the name is raised and insistant by feature PM.